File: pms.py
#A python program to mimic the functionality of a real-world
#property management system(PMS), commonly found in hotels and residential buildings

import time
import datetime
import csv
from csv import writer
import random


#Display main menu to the user
print("\n:::::Python Property Management System:::::\n")
time.sleep(1)
print("Main Menu:")
print("----------------\n")
print("Property Code: PGH620\n")
print("1. Search by Name\n" "2. Search by Confirmation #\n" "3. Search by Room #\n" "4. View In-House Guests\n" "5. View Departures\n" "6. Make a Reservation\n" "7. Cancel Reservation\n" "8. Log-off/Switch User\n")
time.sleep(1) #mimics loading of the system
menuSelect = input("Make a Selection: ")

if menuSelect == "1": 
    nameEntry = input("Please enter Last name: ")
    #opens csv file containing guest records
    with open ('hotelguests.csv', 'rt') as csv_file:
        csv_reader = csv.reader(csv_file, delimiter=',')
        #searches for name entry, displays line
        for row in csv_reader:
            if row[0] == nameEntry:
                print(row)
                #todo: ADD ELIF FOR "GUEST NOT FOUND"
        

elif menuSelect == "2": 
    confEntry = input("Enter Confirmation Number: ")
    #open csv file for guest records/search for confirmation number
    with open ('hotelguests.csv', 'rt') as csv_file:
        csv_reader = csv.reader(csv_file, delimiter=',')
        for column in csv_reader:
            if column[5] == confEntry and confEntry.isdigit():
                print(column)
                #todo: ADD ELIF FOR "GUEST NOT FOUND"

elif menuSelect == "3":
    roomEntry = input("Enter Room Number: ")
    #open csv file for guest records/search for room number
    with open ('hotelguests.csv', 'rt') as csv_file:
        csv_reader = csv.reader(csv_file, delimiter=',')
        for column in csv_reader:
            if column[2] == roomEntry and roomEntry.isdigit():
                print(column)
                #todo: ADD ELIF FOR "GUEST NOT FOUND"


elif menuSelect == "4":
    #open csv file/display all in-house guests
    with open("hotelguests.csv", "r") as csv_file:
        csv_reader = csv.reader(csv_file, delimiter=',')
        for lines in csv_reader:
            print(lines)

elif menuSelect == "5":
    depDate = input("Enter Departure Date: ")
    #open csv file/search for and display guests departing on provided date
    with open ('hotelguests.csv', 'rt') as csv_file:
        csv_reader = csv.reader(csv_file, delimiter=',')
        for column in csv_reader:
            if column[4] == depDate:
                print(column)
                #todo: ADD ELIF FOR NO DEPARTURES FOUND

elif menuSelect == "6": 
    #sets variables for reservation input, to be appended to csv file
    lastName = input("Enter a Last Name: ")
    firstName = input("Enter a First Name: ")
    roomNumber = input("Assign Room Number: ")
    arrivalDate = input("Enter an Arrival Date: ")
    departureDate = input("Enter a Departure Date: ")
    #generates random confirmation number
    confNum = random.randint(150000,160000)
    enterRate = input("Enter Rate Amt: $")
    enterCC = input("Enter a valid Credit Card Number: ")
    enterEXP = input("Enter Expiration Date as mm/yy: ")
    #open csv file in append mode - append/update file with user input for each variable
    with open('hotelguests.csv', 'a', newline='') as csvfile:
        csvwriter = csv.writer(csvfile)
        csvwriter.writerow([lastName, firstName, roomNumber, arrivalDate, departureDate, confNum, enterRate, enterCC, enterEXP])
    print("Uploading Reservation...\n")
    time.sleep(3)
    #prints reservation confirmation as str, using confNum variable(random)
    print("Reservation Uploaded with Confirmation # " + str(confNum))

elif menuSelect == "7":
    print("CXL MENU")
    print("---------")
    print("1. Cancel by Last Name \n")
    print("2. Cancel by Confirmation #\n")
    cancelSelect = input("Select an option: ")
    lines = list()
    if cancelSelect == "1":
        lastName = input("Enter last name: ")
        with open('hotelguests.csv', 'r') as readFile:
            reader = csv.reader(readFile)
            for row in reader:
                lines.append(row)
                for field in row:
                    if field == lastName:
                        lines.remove(row)
        with open('hotelguests.csv', 'w') as writeFile:
            writer = csv.writer(writeFile)
            writer.writerows(lines)
            time.sleep(2)
            print("Reservation removed from PMS.")

    else: 
        confNumber = input("Enter confirmation number: ")
        if cancelSelect == "2":
            with open('hotelguests.csv', 'r') as readFile:
                reader = csv.reader(readFile)
                for row in reader:
                    lines.append(row)
                    for field in row:
                        if field == confNumber:
                            lines.remove(row)

        with open('hotelguests.csv', 'w') as writeFile:
            writer = csv.writer(writeFile)
            writer.writerows(lines)
            time.sleep(2)
            print("Reservation removed from PMS.")

elif menuSelect == 8:
    logoffAnsw = input("LOG OFF? Y/N?")
    if logoffAnsw == 'Y' or 'y':
        print("Logging Off...")
        time.sleep(3)
        print("Logged off at " + time)
    elif logoffAnsw == 'N' or 'n':
        print("Not finished")

File: pms.cpp
//A C++ program to mimic the functionality of a real-world
//property management system(PMS), commonly found in hotels and residential buildings

#include <iostream>



//Display main menu to the user

int main() {
    std::cout << ":::::+Plus Property Management System:::::\n";

    std::cout <<"\nMain Menu:";
    std::cout <<"------------------\n";
    std::cout <<"Property Code: JGH620\n";
    std::cout <<" 1. Search by Name\n 2. Search by Confirmation\n 3. Search by Room\n 4. View In-House Guests\n 5. View Departures\n 6. Make a Reservation\n 7. Cancel a Reservation\n 8. Log-off/Switch User\n";

    int menuSelect = 0;
    std::cout << "Make a Selection: ";
    std::cin >> menuSelect;

    if (menuSelect == 1) {
        char lastName;
        std::cout << "Enter Last Name: ";
        std::cin >> lastName;
    } else if (menuSelect == 2) {
        int confEntry;
        std::cout << "Enter Confirmation Number: ";
        std::cin >> confEntry;
    } else if (menuSelect == 3) {
        int roomNum;
        std::cout << "Enter Room Number: ";
        std::cin >> roomNum;
    } else if (menuSelect == 4) {
        std::cout << "Loading In-House Guests...";
    } else if (menuSelect == 5) {
        int departDate;
        std::cout << "Enter Departure Date(mmddyy): ";
        std::cin >> departDate;
    } else if (menuSelect == 6) {
        std::cout << "Loading Reservation System...";
    } else if (menuSelect == 7) {
        std::cout << "1. Cancel by Last Name\n";
        std::cout << "2. Cancel by Confirmation Number\n";
        int cxlMenuSelect = 0;
        std::cout << "Make a Selection: ";
        std::cin >> cxlMenuSelect;
        if (cxlMenuSelect == 1) {
            char cxlLastName;
            std::cout << "Enter a Last Name: ";
            std::cin >> cxlLastName;
        } else {
            int cxlConf;
            std::cout << "Enter Confirmation Number: ";
            std::cin >> cxlConf;
        }

    } else if (menuSelect == 8) {
        char userAnswer;
        std::cout << "LOG OFF? Y/N?: ";
        std::cin >> userAnswer;
        if (userAnswer == 'Y' || userAnswer == 'y') {
            std::cout << "User Logged Off\n";
        }
        
        
        
    }
}



